src/main/webapp/assets/js/feedScript.js

```diff
@@ -605,16 +614,204 @@ function closeModal() {
+    invalidIds.push('modal-month');
+    errorMessages.push('month must be between 1 and 12');
+  }
+  if (day < 1 || day > monthDayLengths[month - 1] || isBlank(day)) {
+    invalidIds.push('modal-day');
+    errorMessages.push('invalid date');
+  }
+}
+
+/**
+ * Checks if the start and end time are valid.
+ * @param {array} invalidIds
+ * @param {array} errorMessages
+ * @param {array} formElements
+ * @return {void}
+ */
+function validateModalTime(invalidIds, errorMessages, formElements) {
+  const startHour = parseInt(formElements.namedItem('modal-start-hour').value, 10);
+  const startMinute = parseInt(formElements.namedItem('modal-start-minute').value, 10);
+  const startAMorPM = parseInt(formElements.namedItem('start-am-or-pm').value, 10);
+  const endHour = parseInt(formElements.namedItem('modal-end-hour').value, 10);
+  const endMinute = parseInt(formElements.namedItem('modal-end-minute').value, 10);
+  const endAMorPM = parseInt(formElements.namedItem('end-am-or-pm').value, 10);
+
+  // Check if the hours fall between 1-12.
+  if (startHour < 1 || startHour > 12 || isNaN(startHour)) {
+    invalidIds.push('modal-start-hour');
+    errorMessages.push('start hour must be between 1 - 12');
+  }
+  if (endHour < 1 || endHour > 12 || isNaN(endHour)) {
+    invalidIds.push('modal-end-hour');
+    errorMessages.push('end hour must be between 1 - 12');
+  }
+  // Check if the minutes fall between 0 - 60.
+  if (startMinute < 0 || startMinute >= 60 || isNaN(startMinute)) {
+    invalidIds.push('modal-start-minute');
+    errorMessages.push('start minute must be between 00 - 59');
+  }
+  if (endMinute < 0 || endMinute >= 60 || isNaN(startMinute)) {
+    invalidIds.push('modal-end-minute');
+    errorMessages.push('end minute must be between 00 - 59');
+  }
+  // Check if the end time is after the start time.
+  if ((startAMorPM === 'pm' && endAMorPM === 'am') ||
+  (startAMorPM == endAMorPM &&
+  (endHour < startHour || (endHour === startHour && endMinute < startMinute)))) {
+    invalidIds.push('modal-start-hour');
+    invalidIds.push('modal-start-minute');
+    invalidIds.push('start-am-or-pm');
+    invalidIds.push('modal-end-hour');
+    invalidIds.push('modal-end-minute');
+    invalidIds.push('end-am-or-pm');
+    errorMessages.push('end time must be after start time');
+  }
+}
+
+/**
+ * Checks if the given input is blank.
+ * @param {String} input
+ * @return {void}
+ */
+function isBlank(input) {
+  const trimmed = input.trim();
+  if (trimmed) {
+    return false;
+  } else {
+    return true;
+  }
+}
+
+/**
+ * Goes through the form elements. If it is a text input, adds an error if it is blank.
+ * @param {array} invalidIds
+ * @param {array} errorMessages
+ * @param {array} formElements
+ * @return {void}
+ */
+function validateModalText(invalidIds, errorMessages, formElements) {
+  for (let i = 0; i < formElements.length; i++) {
+    if (formElements[i].type === 'text' || formElements[i].type === 'textarea') {
```

Comment:
camelCase for textarea?

Reply:
In Google's style guide, HTML names should be lowercase! https://google.github.io/styleguide/htmlcssguide.html#Capitalization